### PR TITLE
(fix) O3-2284: Fix failing patient registration test

### PR DESCRIPTION
**Pull request by Piumal1999**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Fixed the failing patient registration e2e test:
1. Corrected the patient attribute
2. Implemented a function that waits until the form is loaded, before filling in the data.

## Screenshots

*None.*

## Related Issue
https://issues.openmrs.org/browse/O3-2284

## Other

*None.*

### CODE REVIEW

File: example.env [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # This is an example environment file for configuring dynamic values.
-E2E_BASE_URL=http://localhost:8080/openmrs
+E2E_BASE_URL=https://dev3.openmrs.org/openmrs
```
Comment (ibacher) [inline example.env:2]: I'd prefer if the example setup didn't point at dev3. We need to limit the usage of dev3 for automated testing.

File: e2e/commands/patient-operations.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ export const generateRandomPatient = async (api: APIRequestContext): Promise<Pat
         ],
         attributes: [],
         birthdate: '2020-2-1',
-        birthdateEstimated: true,
+        birthdateEstimated: false,
```
Comment (ibacher) [inline e2e/commands/patient-operations.ts:80]: Is this change necessary?

File: e2e/commands/patient-operations.ts [hunk 3]
```diff
@@ -77,7 +77,7 @@ export const generateRandomPatient = async (api: APIRequestContext): Promise<Pat
         ],
         attributes: [],
         birthdate: '2020-2-1',
```
Comment (ibacher) [inline e2e/commands/patient-operations.ts:79]:
```suggestion
        birthdate: '2020-02-01',
```

File: e2e/commands/patient-operations.ts (hunk 2 above)
Comment (Piumal1999) [reply e2e/commands/patient-operations.ts:80]: Yeah. If we use `birthdayEstimated` as `true`, it will take the patient as a "patient with an unknown birthdate". So when we go to the edit patient page, it will automatically use the "Birthdate unknown" option. 

image

(That change was added with 7dbf922e29a96925edb0bc700b4383f904b8ac6a. And the tests were failing after that)

File: example.env (hunk 1 above)
Comment (Piumal1999) [reply example.env:2]: Yeah, I just wanted to verify that it works with dev3. The patient registration didn't pass when running locally. It fails to load the patient-chart view after creating a patient. I'm still looking into fixing that bug. 

image

I'm going to revert the line after it is resolved.

**Review by jayasanka-sack — COMMENTED** [review]
@Piumal1999 I merged latest changes to your PR and reran tests. All the test cases are passing now.

File: e2e/commands/patient-operations.ts (hunk 2 above)
Comment (Piumal1999) [reply e2e/commands/patient-operations.ts:80]: Yeah, that's right. I changed the date format.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @Piumal1999 !
